src/main/java/com/google/cloud/pontem/FormatGenericRecordAsSpannerMutationFn.java

```diff
@@ -0,0 +1,290 @@
+        } else {
+          throw new RuntimeException(
+              "Unrecognized format in field "
+                  + fieldName
+                  + " for Avro Long: "
+                  + longVal.getClass());
+        }
+      } else if (spannerFieldType.getCode() == Code.FLOAT64
+          && avroFieldType == Schema.Type.DOUBLE) {
+        Object doubleVal = inputRow.get(fieldName);
+        if (doubleVal == null) {
+          // Do nothing with NULL value.
+        } else if (doubleVal instanceof Double) {
+          mutationBuilder.set(fieldName).to((double) inputRow.get(fieldName));
+        } else {
+          throw new RuntimeException(
+              "Unrecognized format in field "
+                  + fieldName
+                  + " for Avro Long: "
+                  + doubleVal.getClass());
+        }
+      } else if (spannerFieldType.getCode() == Code.DATE && avroFieldType == Schema.Type.STRING) {
+        // Sometimes, the row comes back as a Utf8 and sometimes as a String.
+        Object object = inputRow.get(fieldName);
+        if (object == null) {
+          // Do nothing with NULL value.
+        } else if (object instanceof Utf8 || object instanceof String) {
+          mutationBuilder.set(fieldName).to(Date.parseDate((String) object.toString()));
+        } else {
+          throw new RuntimeException(
+              "Unrecognized format in field "
+                  + fieldName
+                  + " for Avro String: "
+                  + object.getClass());
+        }
+      } else if (spannerFieldType.getCode() == Code.TIMESTAMP
+          && avroFieldType == Schema.Type.STRING) {
+        // Sometimes, the row comes back as a Utf8 and sometimes as a String.
+        Object object = inputRow.get(fieldName);
+        if (object == null) {
+          // Do nothing with NULL value.
+        } else if (object instanceof String || object instanceof Utf8) {
+          mutationBuilder.set(fieldName).to(Timestamp.parseTimestamp((String) object.toString()));
+        } else {
+          throw new RuntimeException(
+              "Unrecognized format in field "
+                  + fieldName
+                  + " for Avro String: "
+                  + object.getClass());
+        }
+      } else if (spannerFieldType.getCode() == Code.BYTES && avroFieldType == Schema.Type.BYTES) {
+        Object object = inputRow.get(fieldName);
+        if (object == null) {
+          // Do nothing with NULL value.
+        } else if (object instanceof ByteBuffer) {
+          mutationBuilder.set(fieldName).to(ByteArray.copyFrom((ByteBuffer) object));
+        } else {
+          throw new RuntimeException(
+              "Unrecognized format in field " + fieldName + " for Avro: " + object.getClass());
+        }
+      } else if (spannerFieldType.getCode() == Code.ARRAY && avroFieldType == Schema.Type.ARRAY) {
+        // Parse the Array.
+
+        Schema nestedArrayAvroColumnType = columnSchema.getElementType();
```

Comment:
Could you please extrat the array parsing into it's own method? This becomes somewhat unwieldy to read with all the somewhat similar but slightly different code between the regular type parsing and the array subtype parsing